src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -523,6 +525,18 @@ private async Task RestoreAsync(bool forceRestore, RestoreOperationSource restor
             {
                 await _logger.ShowErrorAsync(Resources.PackageRefNotRestoredBecauseOfNoConsent);
             }
+
+            if (projects.Count() == 1 && projects.FirstOrDefault() is BuildIntegratedNuGetProject)
```

Comment:
I think this violates the perf approach from earlier. 
Raise the event, and the listener can check the assets file existence.

Reply:
You think is worth changing all that code for this case?

Reply:
I think you have 2 options if you want to fix it.

1. Just raise the event. Remember the check needs to be performed by the listener if any, that way we're not needlessly paying the perf cost of checking for the assets file existence. The one thing about this version is that you're raising an event about a restore change...when a restore hasn't been run and it could never change what would've happened.
1. Find a way to schedule a continuation after the solution restore worker has completed. 

The other alternative is to just ignore the feedback, because this race condition is extremely unlikely. It really requires people to open the PM UI super early.

Reply:
@nkolev92 I updated the code, I think this is a better approach, if you think otherwise I think I'll ignore this race condition for now, and fix this in a later pr.